waku/node/v2/wakunode2.nim

```diff
@@ -109,13 +111,14 @@ proc newWakuProto(switch: Switch): WakuProto =
   wakuproto.handler = handle
   return wakuproto
 
-proc run(config: WakuNodeConf) =
+proc run(config: WakuNodeConf) {.async, gcsafe.} =
```

Comment:
Why make it `async` if there is nothing async about it?

Reply:
Or actually, probably better don't `discard switch.start()` at line 158. 
That seems to return a future of futures...
Not sure, but I'm guessing the first future can already fail also. Not sure what to do about the sequence of futures, documentation is lacking.

Reply:
How would you address https://github.com/status-im/nim-waku/issues/55 otherwise? I'm sure there is some other pattern, but that's how it seems to be done in NBC

Reply:
alernative is to make an async wrapper in the example, I started with that but the blocker was GC safety in setupNat, so I ended up attacking it separately. Can probably remove it from here and move the async wrapper to examples if we want. WDYT? How is this different from Eth2Node in NBC?

Reply:
Well, there is only need to make this async if it actually needs to await an action somewhere.
I would guess that `switch.start()` is such action that you want to wait for before continuing with other work. So perhaps `await` that.

Next, `runForever` should be removed. In Waku v1 node this is placed there with the idea that this is the "top call".
You should move this probably in your example itself. And await for the `run` call there.

Reply:
Agree, currently structure is a bit so so. I started attacking it in this PR instead https://github.com/status-im/nim-waku/pull/59